crates/sui-core/src/authority/authority_store.rs

```diff
@@ -692,10 +692,18 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
         self.effects_notify_read
             .notify(transaction_digest, effects.data());
 
-        // Cleanup the lock of the shared objects. This must be done after we write effects, as
-        // effects_exists is used as the guard to avoid re-locking objects for a previously
-        // executed tx. remove_shared_objects_locks.
-        self.remove_shared_objects_locks(transaction_digest, certificate)?;
+        // Clean up the locks of hared objects. This should be done after we write effects, as
+        // effects_exists is used as the guard to avoid re-writing locks for a previously
+        // executed transaction. Otherwise, there can be left-over locks in the tables.
+        // However, the issue is benign because epoch tables are cleaned up for each epoch.
+        let mut deleted_objects = Vec::new();
+        for (object_id, _) in certificate.shared_input_objects() {
+            if !self.object_exists(*object_id)? {
```

Comment:
I know you just moved this code, but it seems like we shouldn't have to do a read here. The TransactionEffects contains the set of objects that have been deleted, so we could intersect that with shared_input_objects. Maybe leave a TODO